File: test/fixtures/sample.ts
// Sample TypeScript module for parser snapshot test
import fs from 'fs';
const lib = require('path');

class Base {}

class MyClass extends Base {
    /** Method doc */
    myMethod(x: number): number {
        return x;
    }

    /** Async method doc */
    async asyncMethod(data: string): Promise<string> {
        return data;
    }

    /** Arrow property doc */
    arrowProp = async (): Promise<void> => {};

    _privateMethod(): void {
        return;
    }
}

// Back-to-back class with no comment - regression check for docstring leak
class NoDocClass extends Base {
    methodNoDoc(): void {
        return;
    }
}

// typed params: required, optional, and default-valued
function greet(name: string, title?: string, punctuation: string = '!') {
    return name;
}

// top-level arrow function with types
const topArrow = (a: number, b: number): number => a + b;

const topAsyncArrow = async (x: number): Promise<number> => x;

function topFunc(x: number): number {
    return x * 2;
}

if (require.main === module) {
    console.log('entry');
}

File: test/fixtures/sample.py
"""Sample Python module for parser snapshot test."""

import os
from pathlib import Path
from .utils import helper, another_helper


class Base:
    """Base class docstring"""
    pass


class MyClass(Base):
    """MyClass docstring"""

    def method(self, x, y=2):
        """Method docstring"""
        return x + y

    async def async_method(self, data):
        """Async method docstring"""
        return data

    def _private_method(self):
        return None

    @staticmethod
    def static_method(a, b):
        return a + b


# Back-to-back class with no docstring - regression check for docstring leak
class NoDocClass(Base):
    def method_no_doc(self):
        return None


def top_function(a, b=2, *args, **kwargs):
    """Top function docstring"""
    return a + b


async def top_async_function(x):
    return x


def _private_top_function():
    return None


if __name__ == "__main__":
    print('run')